### PR TITLE
Set SCCACHE_IDLE_TIMEOUT to INFINITE(0) on Windows

### PR DESCRIPTION
**Pull request by peterjc123**

### CODE REVIEW

File: .jenkins/pytorch/win-test-helpers/build_pytorch.bat [hunk 1]
```diff
@@ -73,6 +73,8 @@ set PATH=%TMP_DIR_WIN%\bin;%PATH%
 :: jenkins has M40, which is 5.2
 if "%TORCH_CUDA_ARCH_LIST%" == "" set TORCH_CUDA_ARCH_LIST=5.2
 
+:: The default sccache idle timeout is 600, which is too short and leads to intermittent build errors.
+set SCCACHE_IDLE_TIMEOUT=0
```
Comment (malfet) [inline .jenkins/pytorch/win-test-helpers/build_pytorch.bat:77]: (Assuming the units here are seconds.) Would increasing it to 20 min help solve intermittent build issues?
And can you please post a few example, where 10 min timeout was not enough?

**Review by ezyang — APPROVED** [review]

File: .jenkins/pytorch/win-test-helpers/build_pytorch.bat (hunk 1 above)
Comment (ezyang) [reply .jenkins/pytorch/win-test-helpers/build_pytorch.bat:77]: I'm guessing that sometimes MSVC just takes a really long to build the files?

**Comment by ezyang** [comment]
I'm going to go ahead and land this, because it doesn't seem like it should do any harm

File: .jenkins/pytorch/win-test-helpers/build_pytorch.bat (hunk 1 above)
Comment (malfet) [reply .jenkins/pytorch/win-test-helpers/build_pytorch.bat:77]: More than 10 min? Wow!

Comment (peterjc123) [reply .jenkins/pytorch/win-test-helpers/build_pytorch.bat:77]: @malfet It doesn't mean building a single source file takes 10 min. It just mean that there have been 10 min without calling it. Considering the linking process of `torch_cpu.dll` takes nearly 7 min in the PR jobs, it is quite normal. As you can see in `.jenkins/pytorch/common.sh`, it is already set to 1200.

Comment (peterjc123) [reply .jenkins/pytorch/win-test-helpers/build_pytorch.bat:77]: > (Assuming the units here are seconds.) Would increasing it to 20 min help solve intermittent build issues?
> And can you please post a few example, where 10 min timeout was not enough?

Sorry, I cannot find it anymore. Before I send this PR, I just randomly skimmed the ci.pytorch.org HUD and found one occurrence of the build failure that is caused by the timeout.

**Comment by peterjc123** [comment]
@malfet I found one occurrence in the binary jobs. Will apply this PR to those jobs later.
```
  [1926/3476] Building CXX object caffe2\CMakeFiles\torch_cpu.dir\operators\channel_stats_op.cc.obj
  FAILED: caffe2/CMakeFiles/torch_cpu.dir/operators/channel_stats_op.cc.obj
  C:\w\b\windows\tmp_bin\sccache-cl.exe  /nologo /TP -DAT_PARALLEL_OPENMP=1 -DCPUINFO_SUPPORTED_PLATFORM=1 -DFMT_HEADER_ONLY=1 -DIDEEP_USE_MKL -DMAGMA_V2 -DMINIZ_DISABLE_ZIP_READER_CRC32_CHECKS -DONNXIFI_ENABLE_EXT=1 -DONNX_ML=1 -DONNX_NAMESPACE=onnx_torch -DTH_BLAS_MKL -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE=1 -D_OPENMP_NOFORCE_MANIFEST -Dtorch_cpu_EXPORTS -Iaten\src -I..\aten\src -I. -I..\ -I..\cmake\..\third_party\benchmark\include -Icaffe2\contrib\aten -I..\third_party\onnx -Ithird_party\onnx -I..\third_party\foxi -Ithird_party\foxi -I..\caffe2\..\torch\csrc\api -I..\caffe2\..\torch\csrc\api\include -I..\caffe2\aten\src\TH -Icaffe2\aten\src\TH -I..\caffe2\..\torch\..\aten\src -Icaffe2\aten\src -Icaffe2\..\aten\src -Icaffe2\..\aten\src\ATen -I..\caffe2\..\torch\csrc -I..\caffe2\..\torch\..\third_party\miniz-2.0.8 -I..\aten\src\TH -I..\aten\..\third_party\catch\single_include -I..\aten\src\ATen\.. -Icaffe2\aten\src\ATen -I..\third_party\miniz-2.0.8 -I..\caffe2\core\nomnigraph\include -I..\c10\.. -Ithird_party\ideep\mkl-dnn\include -I..\third_party\ideep\mkl-dnn\src\..\include -I..\third_party\cpuinfo\include -I..\third_party\fbgemm\include -I..\third_party\fbgemm -I..\third_party\fbgemm\third_party\asmjit\src -I..\third_party\FP16\include -I..\third_party\fmt\include -I..\cmake\..\third_party\googletest\googlemock\include -I..\cmake\..\third_party\googletest\googletest\include -I..\third_party\protobuf\src -IC:\w\b\windows\mkl\include -I..\third_party -I..\cmake\..\third_party\eigen -IC:\w\b\windows\conda\envs\py37\include -IC:\w\b\windows\conda\envs\py37\lib\site-packages\numpy\core\include -I..\cmake\..\third_party\pybind11\include -I..\cmake\..\third_party\cub -IC:\w\b\windows\magma_cuda102_release\include -I"C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2\include" -I..\third_party\ideep\mkl-dnn\include -I..\third_party\ideep\include -I..\caffe2 /DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -openmp -DNDEBUG -DUSE_FBGEMM -DHAVE_AVX_CPU_DEFINITION -DHAVE_AVX2_CPU_DEFINITION /MD /O2 /Ob2 /DNDEBUG /w /bigobj -DNDEBUG   -DCUDA_HAS_FP16=1 -DUSE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD /Z7 /EHsc /DNOMINMAX /wd4267 /wd4251 /wd4522 /wd4838 /wd4305 /wd4244 /wd4190 /wd4101 /wd4996 /wd4275 /bigobj -O2 -DCAFFE2_BUILD_MAIN_LIB -DONNX_BUILD_MAIN_LIB -std:c++14 /showIncludes /Focaffe2\CMakeFiles\torch_cpu.dir\operators\channel_stats_op.cc.obj /Fdcaffe2\CMakeFiles\torch_cpu.dir\ /FS -c ..\caffe2\operators\channel_stats_op.cc
  error: failed to execute compile
  caused by: error reading compile response from server
  caused by: Failed to read response header
  caused by: failed to fill whole buffer
  ninja: build stopped: subcommand failed.
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\w\b\windows\pytorch\setup.py", line 740, in <module>
      build_deps()
    File "C:\w\b\windows\pytorch\setup.py", line 323, in build_deps
      cmake=cmake)
    File "C:\w\b\windows\pytorch\tools\build_pytorch_libs.py", line 62, in build_caffe2
      cmake.build(my_env)
    File "C:\w\b\windows\pytorch\tools\setup_helpers\cmake.py", line 345, in build
      self.run(build_args, my_env)
    File "C:\w\b\windows\pytorch\tools\setup_helpers\cmake.py", line 141, in run
      check_call(command, cwd=self.build_dir, env=env)
    File "C:\w\b\windows\conda\envs\py37\lib\subprocess.py", line 363, in check_call
      raise CalledProcessError(retcode, cmd)
```

**Pull request closed without merging**